src/web/Dockerfile

```diff
@@ -15,14 +15,20 @@ ARG WEB_URL
 ARG SUPABASE_URL
 ARG ANON_KEY
 
+## Base ###########################################################################
 FROM node:16 as base
 
-WORKDIR /frontend
+RUN npm install -g pnpm
 
-COPY . .
+COPY ./app /app
+COPY ./docusaurus /docusaurus
```

Comment:
I think we should have separate stages for each app, otherwise docker will try to build both if one of them changes.

Also, maybe we should install dependencies first in a separate stage, before copying the source into the container?

Reply:
Agree, with this option. It would be better for caching and run our Dockerfile faster.

Maybe we could do that.
```yml
FROM node:16 as base
install only pnpm 

FROM base as dependencies-app
workdir app
Copy package.json of next.js app and pnpm install

FROM base as dependencies-docs
workdir docusaurus
Copy package.json of docs app and pnpm install

FROM base as build-app 
workdir app
Set your ARG for next.js
Copy source code from build context to image
Copy node_modules from dependencies-app
Build it

FROM base as build-docusaurus
workdir docusaurus
Copy source code from build context to image
Copy node_modules from dependencies-docusaurus
Build it

FROM nginx:1.20.2-alpine as deploy
Copy build of next,js from build-app to /var/www/data
Copy build of docusaurus from build-docusaurus to /usr/share/nginx/html
```
@menghif, @manekenpix  what do you think?

Reply:
Yes that's the way I was thinking based on @manekenpix's comment. Unfortunately I am still stuck on the Nginx config (getting 403 for docusaurus).